### PR TITLE
Fix/bid session issues

### PR DESCRIPTION
**Pull request by bohdan-titan**

### CODE REVIEW

**Review by alex-sandrk — APPROVED** [review]

File: proxy-router/internal/blockchainapi/service.go [hunk 1]
```diff
@@ -621,6 +619,26 @@ func (s *BlockchainService) GetSessions(ctx *gin.Context, user, provider common.
 	return mapSessions(ids, sessions, bids), nil
 }
 
+func (s *BlockchainService) GetSessionsIds(ctx *gin.Context, user, provider common.Address, offset *big.Int, limit uint8) ([]common.Hash, error) {
+	var (
+		ids [][32]byte
+		err error
+	)
```
Comment (lsheva) [inline proxy-router/internal/blockchainapi/service.go:626]: you dont need this, use := on the next line

File: proxy-router/internal/blockchainapi/service.go [hunk 2]
```diff
@@ -621,6 +619,26 @@ func (s *BlockchainService) GetSessions(ctx *gin.Context, user, provider common.
...
+
+	ids, err = s.sessionRouter.GetSessionsIdsByUser(ctx, common.HexToAddress(ctx.Query("user")), offset, limit)
```
Comment (lsheva) [inline proxy-router/internal/blockchainapi/service.go:628]:
```suggestion
	ids, err := s.sessionRouter.GetSessionsIdsByUser(ctx, user, offset, limit)
```

File: proxy-router/internal/blockchainapi/service.go [hunk 3]
```diff
@@ -621,6 +619,26 @@ func (s *BlockchainService) GetSessions(ctx *gin.Context, user, provider common.
 	return mapSessions(ids, sessions, bids), nil
 }
 
+func (s *BlockchainService) GetSessionsIds(ctx *gin.Context, user, provider common.Address, offset *big.Int, limit uint8) ([]common.Hash, error) {
```
Comment (lsheva) [inline proxy-router/internal/blockchainapi/service.go:622]: no reason to inject whole gin request here in service. The parameters already parsed in controller

```suggestion
func (s *BlockchainService) GetSessionsIds(ctx context.Context, user, provider common.Address, offset *big.Int, limit uint8) ([]common.Hash, error) {
```

File: proxy-router/internal/repositories/registries/marketplace.go [hunk 4]
```diff
@@ -47,11 +47,17 @@ func (g *Marketplace) PostModelBid(opts *bind.TransactOpts, model common.Hash, p
...
+		if err == nil {
+			bidId, errBid := g.marketplace.GetBidId(&bind.CallOpts{Context: opts.Context}, event.Provider, event.ModelId, event.Nonce)
+			if errBid == nil {
+				return bidId, nil
+			}
+		}
 	}
 
-	return receipt.TxHash, nil
+	return common.Hash{}, nil
```
Comment (lsheva) [inline proxy-router/internal/repositories/registries/marketplace.go:60]: watch out with this, cause most of methods return txhash on first position

File: ui-desktop/src/renderer/src/store/utils/apiCallsHelper.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    
+    const limit = 50;
+    let offset = 0;
+    let sessions: any[] = [];
+    let all = false;
+
+    while (!all) {
+      console.log("Getting session for user: ", user, offset, limit)
+      const ids = await getIds(user, offset, limit);
+      const results = await Promise.all(ids.map(id => getSessioInfo(id)));
```
Comment (lsheva) [inline ui-desktop/src/renderer/src/store/utils/apiCallsHelper.tsx:40]: this may cause errors with public nodes

**Pull request merged**